Tests/kaas/plugin/interface.py

```diff
@@ -0,0 +1,55 @@
+from abc import abstractmethod
+
+
+class KubernetesClusterPlugin():
+    """
+    An abstract base class for custom Kubernetes cluster provider plugins.
+    It represents an interface class from which the api provider-specific
+    plugins must be derived as child classes
+
+    To implement fill the methods `create_cluster` and `delete_cluster` with
+    api provider-specific functionalities for creating and deleting clusters.
+
+    - Implement `create_cluster` and `delete_cluster` methods
+    - Create `__init__(self, config_file=None)` method to handle api specific
+      configurations.
+
+    Example:
+        .. code:: python
+
+          from interface import KubernetesClusterPlugin
+          from apiX_library import cluster_api_class as ClusterAPI
+
+          class PluginX(KubernetesClusterPlugin):
+
+              def __init__(self, config_file=None):
+                  self.config = config_file
+
+              def create_cluster(self):
```

Comment:
This example is missing the parameters that this method has. Same for `delete_cluster` below. Apart from that, nice touch!

Reply:
Thank you! I have added the parameter accordingly.